packages/@react-spectrum/toast/src/ToastContainer.tsx

```diff
@@ -38,13 +39,29 @@ export interface SpectrumToastOptions extends Omit<ToastOptions, 'priority'>, DO
 
 type CloseFunction = () => void;
 
+function wrapInViewTransition<R>(fn: () => R): R {
+  if ('startViewTransition' in document) {
+    let result: R;
+    // @ts-expect-error
+    document.startViewTransition(() => {
+      flushSync(() => {
+        result = fn();
+      });
+    });
+    // @ts-ignore
+    return result;
```

Comment:
I think we're going to run into an issue where we need to await `transition.finished` before return the function call for programatic closes to work (not ideal). Maybe we should revisit the API of returning the close function?

Reply:
has this been resolved?